src/main/java/xyz/aimcup/tournament/mapper/tournament/TournamentMapper.java

```diff
@@ -0,0 +1,13 @@
+package xyz.aimcup.tournament.mapper.tournament;
+
+import java.util.List;
+import org.mapstruct.Mapper;
+import xyz.aimcup.generated.model.TournamentResponseDto;
+import xyz.aimcup.tournament.data.entity.tournament.Tournament;
+
+@Mapper(componentModel = "spring")
+public interface TournamentMapper {
+    TournamentResponseDto toTournamentResponseDto(Tournament tournament);
+
+    List<TournamentResponseDto> toTournamentResponseDtos(List<Tournament> tournaments);
```

Comment:
TournamentResponseDto jest bardzo ogólną nazwą. Przy wielu Dtos dotyczących Tournament bedzie tutaj problem

Reply:
Ja np. w swoim tasku zrobiłem coś takiego: 
`@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface **TournamentCreateMapper** {

    PlayerBasedTournament mapTournamentCreateRequestToPlayerBasedTournament(
        TournamentCreateRequest tournamentCreateRequest);

    InternationalTournament mapTournamentCreateRequestToInternationalTournament(
        TournamentCreateRequest tournamentCreateRequest);

    TeamBasedTournament mapTournamentCreateRequestToTeamBasedTournament(
        TournamentCreateRequest tournamentCreateRequest);
}`


Przy czym tutaj widziałem jak u Ciebie się nazywają metody i chyba zmienię zobie na Podobnie jak u Ciebie jest i zamiast map dam to - czytelniejsze jest faktycznie.


Ale jak widzisz ja tu zamieniłem mapper  na TournamentCreateMapper, bo widzę, że może mieć wiele metod, a nie chcemy raczej klasy mappera która w końcowym efekcie będzie miała 20 metod mapujących i w niektórych jeszcze będą dodatkowe adnotacje. Musimy to jakoś rozdzielać

Reply:
Przemyślałem... zostaw TournamentResponseDto